### ISSUE TITLE
wsgi fails with python 3.4.3

### ISSUE THREAD

**Issue by kidbrax** [skoczen/will#148]
I am brand new to python.  I am trying to install Will and it fails because of the outdated print statemtn when attempting to install wsgiref==0.1.2.  Is there a quick fix for this or do I need to use an older version of Python?  Sorry for the newb question.

```
Collecting wsgiref==0.1.2 (from will->-r requirements.txt (line 1))
  Using cached wsgiref-0.1.2.zip
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 20, in <module>
      File "/private/var/folders/5f/dnt4m_5x0jgf9z4g3hlrbftmnspq8k/T/pip-build-bq0ugkmz/wsgiref/setup.py", line 5, in <module>
        import ez_setup
      File "/private/var/folders/5f/dnt4m_5x0jgf9z4g3hlrbftmnspq8k/T/pip-build-bq0ugkmz/wsgiref/ez_setup/__init__.py", line 170
        print "Setuptools version",version,"or greater has been installed."
                                 ^
    SyntaxError: Missing parentheses in call to 'print'
```

**Comment by skoczen** [comment]
Hey @kidbrax,

Welcome to python and will! 

What you're running into is (unfortunately) common at this point in python's journey.  `3.x` isn't completely backwards compatible with `2.x`, and a number of libraries don't support 3.x yet.

The `0.1.2` version of `wsgiref` is one of them.

There are a few fixes that will get you running, in order of least work to most:

1. Switch over to python2.7 for your will installs.
2. Try to upgrade `wsgiref`  with `pip install --upgrade wsgiref`, and see if the latest version works with your setup, and with will (if it doesn't, you'd notice the http/webhooks stuff not working.
3. If you try 2) and it works, submit a PR here with the upgraded version in requirements.txt.   (You can find out what versions you've got by using `pip freeze`).  

More broadly, it's probably time for me to upgrade *all* the requirements, and refreeze them I've set up #149 for tracking on that work.

Thanks for reaching out on this, and I hope that helps you in getting your will running!

And I promise, python isn't generally like this.  Otherwise, I would have left it years ago. :)

-Steven

**Comment by kidbrax** [comment]
Thanks for the detailed answer.  I'll give it a shot.  It's good to get the support because I've been wondering if I should just stick with Ruby or forge ahead with Python.

**Comment by skoczen** [comment]
Great, keep me posted on how it goes, and ping if you hit any snags.

On Ruby vs Python, I would say that - having spent a year with Ruby and 7 with Python, and knowing lots of colleagues who have spent significant chunks with both - the consensus is that one of the two makes sense to a person.  Rarely both.

I find python clear, expressive, and that it works the way I think.  For my brain, Ruby was an unintuitive mess.  I have close friends who find the opposite to be true.  Take that for whatever that's worth!

**Comment by skoczen** [comment]
Closing this, since #149 covers it more generally!

**Comment by Deepakdubey90** [comment]
Thanks.

**Comment by imanojkumar** [comment]
why this is not available for python 3.9.x version?
